src/stdlib/statistics.jl

```diff
@@ -1,16 +1,78 @@
-export i_mean_sum, i_var_mean_sum, i_normal_logpdf, i_cor_cov
+export i_mean_sum, i_mean, i_sum, i_var_mean_sum, i_normal_logpdf, i_cor_cov
 export VarianceInfo
 
 """
-    i_mean_sum(out!, sum!, x)
+    i_mean_sum(mean_val, sum_val, x)
 
-get the `mean` and `sum` of `x`.
+Get the `mean` and `sum` of `x` in one routine.
+
+```jldoctest; setup=:(using NiLang)
+x = rand(64, 64)
+m, s, x = i_mean_sum(0.0, 0.0, x)
+(m, s) .≈ (mean(x), sum(x))
+
+# output
+(true, true)
+```
+
+See also: [`i_mean`](@ref), [`i_sum`](@ref)
+"""
+@i function i_mean_sum(mean_val, sum_val, x)
+    @inbounds @simd for i in eachindex(x)
```

Comment:
using `eachindex` is because not all arrays are 1-based indexing. A typical case is `OffsetArray`.

```julia
using OffsetArrays
x = OffsetArray(rand(64, 64), -1, -1) # x[0, 0] is the first index
```

Reply:
To use eachindex, you’d better add @invcheckoff before the for loop, otherwise it will check whether the iterable has been changed.

Reply:
I think I made a mistake here, since `x[1]` is indexed linearly, so whether it has offset or not doesn't change the result. Will revert this change.